core/src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java

```diff
@@ -142,16 +142,16 @@ private Scanner executeScanWithIndex(Selection selection, TableMetadata tableMet
 
     builder.expressionAttributeNames(expressionAttributeNames);
 
+    int limit = 0;
     if (selection instanceof Scan) {
       Scan scan = (Scan) selection;
-      if (scan.getLimit() > 0) {
-        builder.limit(scan.getLimit());
```

Comment:
I'm not sure what was the root cause, but I guess this doesn't expectedly work. Is my guess correct?

Reply:
Yes, this doesn’t work as expected. The limit parameter actually functions more like a page size (the number of records in a single request) rather than the total number of records to retrieve.

Reply:
I don't understand why you removed the line `builder.limit(scan.getLimit());`
Could you explain why you don't set the limit on the initial dynamo query only in the case of a ScanWithIndex (`SelectStatementHandler.executeScan()` and `SelectStatementHandler.executeFullScan` still set the limit on the request builder) ?

Reply:
@Torch3333 Instead of using `builder.limit(scan.getLimit());` here, we call the method in the following part (`handleResponse(request.execute(limit));`):
https://github.com/scalar-labs/scalardb/pull/2291/files#diff-08e57801daa3c3c5ed64e1afbf4b043a9b2a581af89d92e64712af5cf051af81R35-R41

So we are actually still calling `builder.limit()` elsewhere.

Reply:
I see, thanks for the explanation. 

Since the limit is set all cases in
 - https://github.com/scalar-labs/scalardb/blob/fe602d4d32e34aaa2806c2d40dbed0d3e9090794/core/src/main/java/com/scalar/db/storage/dynamo/request/QueryRequest.java#L33
 - https://github.com/scalar-labs/scalardb/blob/fe602d4d32e34aaa2806c2d40dbed0d3e9090794/core/src/main/java/com/scalar/db/storage/dynamo/request/ScanRequest.java#L33
 
 Do you think we could safely remove these two lines?
 - https://github.com/scalar-labs/scalardb/blob/fe602d4d32e34aaa2806c2d40dbed0d3e9090794/core/src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java#L175
 - https://github.com/scalar-labs/scalardb/blob/fe602d4d32e34aaa2806c2d40dbed0d3e9090794/core/src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java#L199

Reply:
Ah yes, good catch! I forgot to remove those. I’ll fix them. thanks!

Reply:
Fixed in 244babbb226362abd0517b981d10931bea0321ac. Thanks.